# DP: Description: Only parse three parts of a kernel version number.
# DP: Related bugs: 218546
# DP: Author: Daniel Jacobowitz <dan@debian.org>
# DP: Upstream status: Debian-specic
# DP: Status details: Rejected by Ulrich, needed for Debian.
# DP:       http://sources.redhat.com/ml/libc-alpha/2003-11/msg00025.html
# DP: Date: 2003-11-01

2003-11-01  Daniel Jacobowitz  <dan@debian.org>

	* sysdeps/unix/sysv/linux/dl-osinfo.h (DL_SYSDEP_OSCHECK): Don't
	parse more than three parts of the version number.

--- sysdeps/unix/sysv/linux/dl-osinfo.h.orig
+++ sysdeps/unix/sysv/linux/dl-osinfo.h
@@ -126,7 +126,7 @@
       version <<= 8;
       version |= here;
 
-      if (*cp++ != '.')
+      if (*cp++ != '.' || parts == 3)
 	/* Another part following?  */
 	break;
     }
